app/static/src/app/components/WodinPlot.vue

```diff
@@ -62,40 +78,94 @@ export default defineComponent({
                 store.commit(`graphs/${GraphsMutation.SetYAxisRange}`, yRange);
             }
         };
 
-        const relayout = async (event: PlotRelayoutEvent) => {
+        const defaultLayout = (): Partial<Layout> => {
+            // Get generic layout, which will be modifed dynamically as required
+            const result = {
+                margin: { ...margin },
+                xaxis: { title: "Time" },
+                yaxis: { type: yAxisType.value }
+            };
+            if (legendWidth.value) {
+                result.margin.r = legendWidth.value;
+            }
+            return result;
+        };
+
+        const preserveYAxisRange = (layout: Partial<Layout>): Partial<Layout> => {
+            // When updating plot view in response to some data change or event, retain Y axis range in layout
+            // either from the locked range, or from the last range user zoomed to.
+            // (Locked range will survive re-mount and tab change, the last range from zoom will not)
+            const result = { ...layout };
+            if (lockYAxis.value) {
+                result.yaxis = {
+                    ...result.yaxis,
+                    range: [...yAxisRange.value],
+                    autorange: false
+                };
+            } else if (lastYAxisFromZoom.value) {
+                result.yaxis = {
+                    ...result.yaxis,
+                    ...lastYAxisFromZoom.value
+                };
+            }
+            return result;
+        };
+
+        const updateXAxisRange = async (xAxis: Partial<LayoutAxis>) => {
             let data;
-            if (event["xaxis.autorange"] === true) {
+            if (xAxis.autorange) {
                 data = baseData.value;
             } else {
-                const t0 = event["xaxis.range[0]"];
-                const t1 = event["xaxis.range[1]"];
-                if (t0 === undefined || t1 === undefined) {
-                    return;
-                }
-                data = props.plotData(t0, t1, nPoints);
+                data = props.plotData(xAxis.range![0], xAxis.range![1], nPoints);
             }
 
-            const layout: Partial<Layout> = {
-                margin,
-                uirevision: "true",
-                xaxis: { title: "Time", autorange: true },
-                yaxis: { autorange: true, type: yAxisType.value }
-            };
+            const layout = preserveYAxisRange(defaultLayout());
 
             const el = plot.value as HTMLElement;
             await react(el, data, layout, config);
         };
 
+        const axisFromEvent = (event: PlotRelayoutEvent, axisLetter: "x" | "y"): Partial<LayoutAxis> => {
+            return {
+                autorange: event[`${axisLetter}axis.autorange`] || false,
+                range: [event[`${axisLetter}axis.range[0]`], event[`${axisLetter}axis.range[1]`]]
+            };
+        };
+
+        const relayout = async (event: PlotRelayoutEvent) => {
```

Comment:
```suggestion
        const reLayout = async (event: PlotRelayoutEvent) => {
```

Reply:
I think I'm going to leave this as is, since this is a Plotly word, and it doesn't capitalise the l wherever 'relayout` appears, so I think it would be inconsistent if we did.